package sec03_throwsexception.EX01_ThrowsException_1;

//1. 하위 메서드에 직접 예외를 처리할 때
class A {
	void abc() {
		bcd();
	}
	void bcd() {
		try {
			Thread.sleep(1000);// 일반 예외 : InterruptedException
		}catch(InterruptedException e) {
			// 예외 처리 구문
		}
	}
}

class B {
	void abc() {
		try {
			bcd();
		}catch(InterruptedException e) {
			//	예외 처리 구문
		}
	}
	void bcd() throws InterruptedException {
		Thread.sleep(1000);
	}
}

public class ThrowsException_1 {
	public static void main(String[] args) {

	}
}
